build-tools/packages/build-cli/src/library/releasePrepChecks.ts

```diff
@@ -228,3 +234,51 @@ export const CheckNoUntaggedAsserts: CheckFunction = async (
 		};
 	}
 };
+
+/**
+ * Checks that packages with layer compatibility metadata have up-to-date generation files.
+ * This check is lenient - packages without metadata or generation files are skipped.
+ */
+export const CheckLayerCompatGeneration: CheckFunction = async (
```

Comment:
There is a `checkCompatLayerGeneration` function in `handlers/checkFunctions.ts`. is it possible to resuse some of the logic in both these? For instance, the message with the fix command is useful and should be added to `checkCompatLayerGeneration` function.